### ISSUE TITLE
CORS error sending registration form

### ISSUE THREAD

**Issue by seporterfield** [Monstarrrr/rebutify#121]
```
[HTTP/1.1 101 Switching Protocols 2ms]

# Intercepted request: 
Object { transitional: {…}, adapter: (3) […], transformRequest: (1) […], transformResponse: (1) […], timeout: 0, xsrfCookieName: "XSRF-TOKEN", xsrfHeaderName: "X-XSRF-TOKEN", maxContentLength: -1, maxBodyLength: -1, env: {…}, … }
api.ts:35:12
XHROPTIONS
http://127.0.0.1:8000/auth/users/
CORS Missing Allow Header

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8000/auth/users/. (Reason: header ‘access_token’ is not allowed according to header ‘Access-Control-Allow-Headers’ from CORS preflight response).

XHRPOST
http://127.0.0.1:8000/auth/users/

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8000/auth/users/. (Reason: CORS request did not succeed). Status code: (null).

The resource at “http://localhost:3000/_next/static/media/c9a5bc6a7c948fb0-s.p.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly. register
```

This is the browser error I get when filling out a valid registration form and hitting send.

**Comment by seporterfield** [comment]
This may have been because I had local storage in my browser that was still saved between updates.

Deleting all local storage variables in my browser in developer tools fixed the issue.

**Comment by seporterfield** [comment]
I replicated this again. Just want to check whether this is actually an issue or not.

1. Unregistered user registers, success, clicks email link, success, login, success.

Now there are some items in the browser's local storage.

2. Wipe user from DB.
3. The now again unregistered user registers, ... CORS errors!

```
XHROPTIONS
http://127.0.0.1:8000/auth/users/
CORS Missing Allow Header

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8000/auth/users/. (Reason: header ‘access_token’ is not allowed according to header ‘Access-Control-Allow-Headers’ from CORS preflight response).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:8000/auth/users/. (Reason: CORS request did not succeed). Status code: (null).

XHRPOST
http://127.0.0.1:8000/auth/users/
```

**Comment by seporterfield** [comment]
Guess:

```ts
if (access_token !== 'null' && access_token !== 'undefined') {
      req.headers.access_token = localStorage.getItem('access_token')
    }
```
Needs `req.headers.token` instead of `req.headers.access_token` to match what djoser is expecting?
@Monstarrrr

**Comment by seporterfield** [comment]
After some investigation, the main problem seems to be that we're sending `access_token` in the headers, when really certain routes require certain headers and bodies.

For example,
- `/auth/users` doesn't take special headers at all.
- `/auth/jwt/create` doesn't take special headers, just user/pass in body

**Comment by Monstarrrr** [comment]
I didn't move it to done, bug.

**Comment by Monstarrrr** [comment]
> registered user goes to registration form again immediately after registering and should get redirected elsewhere or given a message like "user already registered" but instead gets CORS errors  

Will be addressed by #110

**Comment by Monstarrrr** [comment]
> someone on the same computer wants to register another account but instead gets CORS errors  

Could not reproduce, is this bug still happening on ``main`` ?

**Comment by Monstarrrr** [comment]
> > someone on the same computer wants to register another account but instead gets CORS errors
> 
> Could not reproduce, is this bug still happening on `main` ?

Update: Got the error, will fix as part of #111 's tests

**Comment by Monstarrrr** [comment]
> > someone on the same computer wants to register another account but instead gets CORS errors
> 
> Could not reproduce, is this bug still happening on `main` ?

### Update  
I could reproduce this case:  
> 1. Unregistered user registers, success, clicks email link, success, login, success.
>  2. Wipe user from DB.
> 3. The now again unregistered user registers, ... CORS errors!

**Comment by Monstarrrr** [comment]
> > registered user goes to registration form again immediately after registering and should get redirected elsewhere or given a message like "user already registered" but instead gets CORS errors
> 
> Will be addressed by #110

Decided against fixing in #110 to keep PRs small as the fix will require some extra feature

**Comment by Monstarrrr** [comment]
- [x] Find error origin
Header ``access_token`` is not allowed according to header ‘Access-Control-Allow-Headers’ in the back-end
- [ ] Add either refresh or access token in ``Authorization`` header (depending on if it expired) instead 

Blocked by #162
